src/MANYBODY/pair_eam_cd.cpp

```diff
@@ -502,22 +490,19 @@ void PairEAMCD::read_h_coeff(char *filename)
     // Open potential file
 
     FILE *fptr;
-    char line[MAXLINE];
-    char nextline[MAXLINE];
-    int convert_flag = unit_convert_flag;
+    char line[2][MAXLINE];
+    int convert_flag = unit_convert_flag, toggle = 0;
     fptr = utils::open_potential(filename, lmp, &convert_flag);
     if (fptr == nullptr)
-      error->one(FLERR,"Cannot open EAMCD potential file {}",
-                                   filename);
+      error->one(FLERR,"Cannot open EAMCD potential file {}", filename);
 
     // h coefficients are stored at the end of the file.
     // Skip to last line of file.
 
-    while (fgets(nextline, MAXLINE, fptr) != nullptr) {
-      strcpy(line, nextline);
-    }
+    while (fgets(line[toggle], MAXLINE, fptr) != nullptr)
```

Comment:
@rbberger please have a look at this change. This avoids the additional copy of the line by alternating between two buffers, but I am not certain whether doing it this way is obvious enough. I should probably add some comment. We could also just allocate two buffers with `new` and swap them, for example. Might be easier to read. This was just the first thought that came to me based on the existing code.

Reply:
New version is implemented as discussed. had to deal with a minor complication (`fread()` does not 0-terminate) but it now works. It also was tested successfully with a potential file lacking the final newline.